narwhals/_pandas_like/dataframe.py

```diff
@@ -167,16 +167,47 @@ def filter(
 
     def with_columns(
         self,
-        *exprs: IntoPandasExpr | Iterable[IntoPandasExpr],
+        *exprs: IntoPandasExpr,
         **named_exprs: IntoPandasExpr,
     ) -> Self:
+        from narwhals._pandas_like.expr import PandasExpr
```

Comment:
Very nit: I wonder if it's worth to import it on top level. I would expect `with_columns` to be a pretty common method

Reply:
yup, this works, thanks! for some reason i thought i was gonna get a circular import

in general these imports should be quite cheap, if we can reduce the circular imports that's probably a good thing